client/src/App.js

```diff
@@ -1,52 +1,98 @@
+    this.state = {
+      isRoot: 1,
+      isPreview: 0
 
-function App() {
-  const courses = [
-    {
-      code: "CS F211",
-      name: "DSA",
-      lecture: [
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
-      ],
-      tutorial: [
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
-      ],
-      lab: [
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
-      ],
-      midsem: "3 Mar",
-      compre: "4 May"
-    },
-    {
-      code: "CS F212",
-      name: "OOPS",
-      lecture: [
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
-      ],
-      tutorial: [
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
-      ],
-      lab: [
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
-        { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
-      ],
-      midsem: "3 Mar",
-      compre: "4 May"
     }
-  ];
+    this.courses = [
+      {
+        code: "CS F211",
+        name: "DSA",
+        lecture: [
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
+        ],
+        tutorial: [
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
+        ],
+        lab: [
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
+        ],
+        midsem: "3 Mar",
+        compre: "4 May"
+      },
+      {
+        code: "CS F212",
+        name: "OOPS",
+        lecture: [
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
+        ],
+        tutorial: [
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
+        ],
+        lab: [
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" },
+          { id: 1, teacher: "abc", hours: " 6 7", room: "ht", days: "hk" }
+        ],
+        midsem: "3 Mar",
+        compre: "4 May"
+      }
+        ];
+      this.sendPrevOrBack = this.sendPrevOrBack.bind(this);
+
+  }
+  sendPrevOrBack()
+  {
+    this.setState(state => ({
+      isPreview: !state.isPreview
+    }));
+    this.setState(state =>({
+      isRoot: !state.isRoot
+    }));
```

Comment:
`this.setState` works with a function too? This is new.
Anyways, these two calls, can be made into a single call, I think

Reply:
I wouldn't do that if we need more than 2 different views. This is a temporary

Reply:
Made another function. Just check it out